test/controller_SUITE.erl

```diff
@@ -1,18 +1,29 @@
 -module(controller_SUITE).
 -compile([export_all, nowarn_export_all]).
 
+-include_lib("proper/include/proper.hrl").
 -include_lib("stdlib/include/assert.hrl").
 
 all() ->
     [
+     {group, distribute},
      {group, all_tests}
     ].
 
 groups() ->
     [
+     {distribute, [parallel], distribute()},
```

Comment:
please add one more test:
* The number of items in the resulting KV list is equal to the number of non-zero items in the input KV list or requested amount (equal to the lower number)